### PR TITLE
qemu (QEMU): split guest agent (qemu-guest-agent) from main package

### PR DESCRIPTION
**Pull request by stdmnpkg**
Topic Description
-----------------

- qemu-guest-agent: new, 9.0.1-3, split from qemu

Package(s) Affected
-------------------

- qemu: 9.0.1-3
- qemu-aarch64-static: 9.0.1-3
- qemu-alpha-static: 9.0.1-3
- qemu-arm-static: 9.0.1-3
- qemu-armeb-static: 9.0.1-3
- qemu-cris-static: 9.0.1-3
- qemu-guest-agent: 9.0.1-3
- qemu-i386-static: 9.0.1-3
- qemu-loongarch64-static: 9.0.1-3
- qemu-m68k-static: 9.0.1-3
- qemu-microblaze-static: 9.0.1-3
- qemu-microblazeel-static: 9.0.1-3
- qemu-mips-static: 9.0.1-3
- qemu-mips64-static: 9.0.1-3
- qemu-mips64el-static: 9.0.1-3
- qemu-mipsel-static: 9.0.1-3
- qemu-mipsn32-static: 9.0.1-3
- qemu-mipsn32el-static: 9.0.1-3
- qemu-nios2-static: 9.0.1-3
- qemu-or32-static: 9.0.1-3
- qemu-ppc-static: 9.0.1-3
- qemu-ppc64-static: 9.0.1-3
- qemu-ppc64le-static: 9.0.1-3
- qemu-riscv32-static: 9.0.1-3
- qemu-riscv64-static: 9.0.1-3
- qemu-s390x-static: 9.0.1-3
- qemu-sh4-static: 9.0.1-3
- qemu-sh4eb-static: 9.0.1-3
- qemu-sparc-static: 9.0.1-3
- qemu-sparc32plus-static: 9.0.1-3
- qemu-sparc64-static: 9.0.1-3
- qemu-user-static: 9.0.1-3
- qemu-x86-64-static: 9.0.1-3

Security Update?
----------------

No

Build Order
-----------

```
#buildit qemu
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

### CODE REVIEW

File: app-virtualization/qemu/01-shared/build [hunk 1]
```diff
@@ -125,10 +125,18 @@ pushd "${SRCDIR}/src.system/build"
 		--extra-cflags="${CFLAGS}"
 
 	abinfo "Building..."
-	make
+	make -j
```
Comment (KexyBiscuit) [inline app-virtualization/qemu/01-shared/build]: Really? No OOM or racing conditions on all architectures observed?

File: app-virtualization/qemu/34-guest-agent/defines [hunk 2]
```diff
@@ -0,0 +1,10 @@
+PKGNAME=qemu-guest-agent
+PKGSEC=admin
+PKGDEP=""
+BUILDDEP=""
+PKGDES="QEMU guest agent"
+
+AB_FLAGS_SPECS=0
+
+PKGBREAK="qemu<=9.0.1-2"
+PKGREP="qemu<=9.0.1-2"
```
Comment (KexyBiscuit) [inline app-virtualization/qemu/34-guest-agent/defines]: Missing EOL.

File: app-virtualization/qemu/spec
Comment (KexyBiscuit) [inline app-virtualization/qemu/spec:1]: Should `qemu` depends on, or recommends, `qemu-guest-agent`?

**Review by KexyBiscuit — DISMISSED** [review]
Not sure if this is the right way of splitting packages... Just a few format linting for now.

File: app-virtualization/qemu/spec
Comment (stdmnpkg) [reply app-virtualization/qemu/spec:1]: no, guest agent should be installed on virtual machine

File: app-virtualization/qemu/01-shared/build (hunk 1 above)
Comment (MingcongBai) [reply app-virtualization/qemu/01-shared/build]: Plus this should not be needed at all, Autobuild defines `$MAKEFLAGS` internally (`$(nproc) + 1`).

File: app-virtualization/qemu/01-shared/build [hunk 3]
```diff
@@ -139,7 +147,7 @@ pushd "${SRCDIR}/src.user/build"
 		--extra-cflags="${CFLAGS}"
 
 	abinfo "Building static user emulation tools"
-	make
+	make -j
```
Comment (MingcongBai) [inline app-virtualization/qemu/01-shared/build]: Please drop the `-j`.

File: app-virtualization/qemu/01-shared/build [hunk 4]
```diff
@@ -125,10 +125,18 @@ pushd "${SRCDIR}/src.system/build"
...
 	abinfo "Building..."
-	make
+	make -j
 
 	abinfo "Installing to PKGDIR..."
 	make DESTDIR="${PKGDIR}" install
+
+	abinfo "Move QEMU guest agent out of package..."
+	mkdir -p "$SRCDIR"/guest-agent
+	mv -v "$PKGDIR"/usr/bin/qemu-ga "$SRCDIR"/guest-agent
```
Comment (MingcongBai) [inline app-virtualization/qemu/01-shared/build]: Consider merging some of the `mv`s, as only the last operand would be considered "target."

File: app-virtualization/qemu/34-guest-agent/defines [hunk 5]
```diff
@@ -0,0 +1,10 @@
+PKGNAME=qemu-guest-agent
+PKGSEC=admin
+PKGDEP=""
+BUILDDEP=""
+PKGDES="QEMU guest agent"
```
Comment (MingcongBai) [inline app-virtualization/qemu/34-guest-agent/defines]: Helper daemon for information exchange between the host and guest machines

File: app-virtualization/qemu/34-guest-agent/defines [hunk 6]
```diff
@@ -0,0 +1,10 @@
+PKGNAME=qemu-guest-agent
+PKGSEC=admin
+PKGDEP=""
```
Comment (MingcongBai) [inline app-virtualization/qemu/34-guest-agent/defines]: Yeah this is probably not right...

File: app-virtualization/qemu/34-guest-agent/defines [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+BUILDDEP=""
```
Comment (MingcongBai) [inline app-virtualization/qemu/34-guest-agent/defines]: Just drop this line.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Comment by stdmnpkg** [comment]
works on my x86 virtual machine, host is proxmox ve 8.2.4

use `systemctl start qemu-ga-virtio@org.qemu.guest-agent.0` to start service

image

**Comment by stdmnpkg** [comment]
renamed systemd unit file to `qemu-guest-agent@.service` to keep consistent with package name and service name of other distros.

filled install section, set `DefaultInstance=org.qemu.guest_agent.0`

File: app-virtualization/qemu/01-shared/build [hunk 8]
```diff
@@ -129,6 +129,16 @@ pushd "${SRCDIR}/src.system/build"
...
+
+	abinfo "Move QEMU guest agent out of package..."
+	mkdir -p "$SRCDIR"/guest-agent
+	mv -v \
+		"$PKGDIR"/usr/bin/qemu-ga \
+		"$PKGDIR"/usr/share/doc/qemu/interop/qemu-ga.html \
+		"$PKGDIR"/usr/share/doc/qemu/interop/qemu-ga-ref.html \
+		"$PKGDIR"/usr/share/man/man8/qemu-ga.8 \
+		"$PKGDIR"/usr/share/man/man7/qemu-ga-ref.7 \
+		"$SRCDIR"/guest-agent # distination
```
Comment (MingcongBai) [inline app-virtualization/qemu/01-shared/build]: destination *

File: app-virtualization/qemu/34-guest-agent/build [hunk 9]
```diff
@@ -0,0 +1,16 @@
+abinfo 'Removing $PKGDIR ...'
+rm -rfv "$PKGDIR"
```
Comment (MingcongBai) [inline app-virtualization/qemu/34-guest-agent/build]: This is not necessary as Autobuild will clean up `$PKGDIR` in sequential builds.

File: app-virtualization/qemu/34-guest-agent/overrides/usr/lib/systemd/system/qemu-guest-agent.service [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+After=dev-virtio\x2dports-org.qemu.guest_agent.0.device
+
+[Service]
+Type=simple
+ExecStart=-/usr/bin/qemu-ga
+StandardError=journal
+Restart=always
+RestartSec=0
+
+[Install]
```
Comment (MingcongBai) [inline app-virtualization/qemu/34-guest-agent/overrides/usr/lib/systemd/system/qemu-guest-agent.service]: Why is this section empty?

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: app-virtualization/qemu/34-guest-agent/overrides/usr/lib/systemd/system/qemu-guest-agent.service (hunk 10 above)
Comment (leavelet) [reply app-virtualization/qemu/34-guest-agent/overrides/usr/lib/systemd/system/qemu-guest-agent.service]: The service will be triggered by UDev rule `/usr/lib/udev/rules.d/99-qemu-guest-agent.rules`

**Comment by stdmnpkg** [comment]
now using udev rule to start systemd service, no longer require `systemctl enable`

works on my x86 machine

image

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**